File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/secretsmanager/SecretValue.java
package uk.nhs.nhsx.core.aws.secretsmanager;

import uk.nhs.nhsx.core.ValueType;

public class SecretValue extends ValueType<SecretValue> {

    private SecretValue(String value) {
        super(value);
    }

    public static SecretValue of(String secret) {
        return new SecretValue(secret);
    }

    @Override
    public String toString() {
        return "Secret{secret=****}";
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/secretsmanager/SecretValue.kt
package uk.nhs.nhsx.core.aws.secretsmanager

import dev.forkhandles.values.Maskers.hidden
import dev.forkhandles.values.NonBlankStringValueFactory
import dev.forkhandles.values.StringValue

class SecretValue private constructor(value: String) : StringValue(value, hidden()) {
    companion object : NonBlankStringValueFactory<SecretValue>(::SecretValue)
}
